### PR TITLE
The great refactoring

### PR DESCRIPTION
**Pull request by apirogov**
Minor:

* moved and renamed some things (too many things were named `core` making it meaningless)
* `path.exists() -> path.is_file()` everywhere (`exists` also accepts a directory)

Major:
* refactored and streamlined project writers (much more code reuse, less ad-hoc overrides)

**This will simplify to come up with a coherent target-independent person merging (separate PR)**

### CODE REVIEW

**Comment by apirogov** [comment]
I think to avoid pains going forward, maybe you should first merge this, then I start working on the Person merge @mustafasoylu

File: src/somesy/core/writer.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+        Implement this in a way that will carefully
+        update the target file with new metadata
+        without destroying its other contents or structure.
+        """
+
+    def _get_property(self, key: Union[str, List[str]]) -> Optional[Any]:
+        """Get a property from the data.
+
+        Override this to e.g. rewrite the retrieved key
+        (e.g. if everything relevant is in some suboject).
```
Comment (mustafasoylu) [inline src/somesy/core/writer.py:88]: typo: subobject

File: src/somesy/core/writer.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            if curr is None:
+                return None
+
+        return curr
+
+    def _set_property(self, key: Union[str, List[str]], value: Any) -> None:
+        """Set a property in the data.
+
+        Override this to e.g. rewrite the retrieved key
+        (e.g. if everything relevant is in some suboject).
```
Comment (mustafasoylu) [inline src/somesy/core/writer.py:104]: typo: subobject

**Review by mustafasoylu — CHANGES_REQUESTED** [review]
You made an incredible job here! I took a model where I separated functions into files, but this seems tidier. Just found a typo, I will approve right after that.

**Review by mustafasoylu — APPROVED** [review]
LGTM

**Pull request merged**